ProQuest.js

```diff
@@ -120,79 +120,96 @@ function fetchEmbeddedPdf(url, item, callback) {
 }
 
 function getSearchResults(doc, checkOnly) {
-	var tabs = doc.getElementsByClassName('tabContent');
 	var root;
-	if (tabs.length) {
-		for (var i = 0; i < tabs.length; i++) {
-			if (tabs[i].offsetHeight) {
-				if (Zotero.isBookmarklet && tabs[i].id != 'allResults-content') return false;
-				root = tabs[i].getElementsByClassName('resultListContainer')[0];
+	var tabs = doc.getElementsByClassName('tabContent');
+	for (var i = 0; i < tabs.length; i++) {
+		if (tabs[i].offsetHeight) {
+			if (Zotero.isBookmarklet && tabs[i].id != 'allResults-content') return false;
+			root = tabs[i].getElementsByClassName('resultListContainer')[0];
+			break;
+		}
+	}
+	if (!root) {
+		var elements = doc.getElementsByClassName('resultListContainer');
+		for (var i=0; i<elements.length; i++) {
+			if (elements[i] && elements[i].offsetHeight>0) {
+				root = elements[i];
 				break;
 			}
 		}
-	} else {
-		root = doc.getElementsByClassName('resultListContainer')[0];
-		if (root && !root.offsetHeight) return false; // Not sure if this can actually happen
 	}
-
-	if (!root) return false;
+	if (!root) {
+		Z.debug("No root found");
+		return false;
+	}
 
 	var results = doc.getElementsByClassName('resultItem');
-	//ZU.xpath(root, './/a[contains(@class,"previewTitle") or contains(@class,"resultTitle")]');
-	
+
 	var items = {}, found = false;
 	for(var i=0, n=results.length; i<n; i++) {
-		var title = results[i].getElementsByClassName('resultTitle')[0]
-			|| results[i].getElementsByClassName('previewTitle')[0];
-		if (!title || title.nodeName != 'A') continue;
-		
-		if (checkOnly) return true;
-		found = true;
-		
-		items[title.href] = title.textContent;
+		if (results[i].offsetHeight>0) { //restrict to the active results only
+			var title = results[i].getElementsByClassName('resultTitle')[0]
+				|| results[i].getElementsByClassName('previewTitle')[0];
+			if (!title || title.nodeName != 'A') continue;
+			
+			if (checkOnly) return true;
+			found = true;
+			
+			items[title.href] = title.textContent;
+		}
 	}
 		
 	return found ? items : false;
 }
 
+var map = {
+	"newspaperArticle" : "newspaperArticle",
+	"jnlArticle" : "journalArticle", //also magazineArticles
+	"dissertationPublished" : "thesis",
+	"reportSerial" : "report",
+	"workingPaper" : "report",
+	"book" : "book",
+	"webPage" : "webpage",
+	"blog" : "blogPost",
+	"report" : "report",
+	"audioVideo" : "videoRecording",
+	"default" : "conferencePaper" //also default = manuscript?, referenceWork, encyclopedia
+}
+
 function detectWeb(doc, url) {
 	initLang(doc, url);
-
-	followLink = false;
-
+	
+	followLink = true;
+	var active = doc.getElementsByClassName('active');
+	for (var i=0; i<active.length; i++) {
+		if ( active[i].id.indexOf("tab-AbstractRecord-null") > -1) {
+			followLink = false;
+		}
+	}
```

Comment:
So basically,
```
var abstractTab = document.getElementById('tab-AbstractRecord-null'); // Seems like that null is a bug and it might change at some point
followLink = !(abstractTab && abstractTab .classList.contains('active'));
```